src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -921,4 +921,45 @@ class DeltaSuite extends QueryTest
       checkAnswer(spark.table("sc15200test"), Seq.empty)
     }
   }
+
+  test("support for setting dataChange to false") {
+    withTempDir { tempDir =>
+      if (tempDir.exists()) {
+        assert(tempDir.delete())
+      }
+      val deltaLogUrl = DeltaLog.forTable(spark, tempDir).snapshot.path
+
+      spark.range(100)
+        .repartition(4)
+        .write
+        .format("delta")
+        .save(tempDir.toString)
+
+      val df = spark.read.format("delta").load(tempDir.toString)
+
+      df
+        .repartition(2)
+        .write
+        .format("delta")
+        .mode("overwrite")
+        .option("dataChange", "false")
+        .save(tempDir.toString)
+
+      val removesDataChange = spark.read
+        .json(deltaLogUrl + "/00000000000000000001.json")
+        .where("remove IS NOT NULL").select($"remove.dataChange")
+        .as[Boolean]
+        .collect()
+        .toSeq
+
+      val addsDataChange = spark.read
+        .json(deltaLogUrl + "/00000000000000000001.json")
+        .where("add IS NOT NULL").select($"add.dataChange")
+        .as[Boolean]
+        .collect()
+        .toSeq
+
+      assert(removesDataChange ++ addsDataChange == Seq(false, false, false, false, false, false))
+    }
+  }
```

Comment:
we should also add a test that makes sure by default, removes and adds leave this as true.

Reply:
ok, I will add this test